fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -808,6 +808,26 @@ public Void visitFunctionCall(FunctionCallExpr node, Scope scope) {
                             fnName + " requires second parameter must be greater than 0");
                 }
                 fn = Expr.getBuiltinFunction(fnName, argumentTypes, Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+            } else if (fnName.equals(FunctionSet.PERCENTILE_DISC)) {
+                argumentTypes[1] = Type.DOUBLE;
+                fn = Expr.getBuiltinFunction(fnName, argumentTypes, Function.CompareMode.IS_IDENTICAL);
```

Comment:
Because percentile_disc's first parameter supports all sortable types, IS_SUPERTYPE_OF or IS_NONSTRICT_SUPERTYPE_OF will find wrong function. So I set the second parameter to double Explicitly to get the correct function, and check the second parameter later in AggFunction analyse